File: backend/01. java/spring_market/spring_market/module-domain/src/main/java/io/github/lokiwooooo/domain/user/repository/UserRepository.java
package io.github.lokiwooooo.domain.user.repository;

import io.github.lokiwooooo.domain.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, String> {
    Optional<User> findByName(final String userName);
}

File: backend/02. kotlin/02.project/spring-boot/library-app/src/main/kotlin/com/group/libraryapp/domain/user/UserRepository.kt
package com.group.libraryapp.domain.user

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import java.util.*

interface UserRepository : JpaRepository<User, Long>, UserRepositoryCustom {

    fun findByName(username: String): User?

    // 같은데이터는 제거함 - 중복제거 distinct
//    @Query("select distinct u from User u left join fetch u.userLoanHistories")
//    fun findAllWithHistories(): List<User>
}